apis/apiextensions/v1/composition_environment.go

```diff
@@ -79,6 +91,26 @@ type EnvironmentSourceSelector struct {
 	MatchLabels []EnvironmentSourceSelectorLabelMatcher `json:"matchLabels,omitempty"`
 }
 
+// Policy represents policies of EnvironmentSourceReference instance.
+type Policy struct {
```

Comment:
https://github.com/crossplane/crossplane-runtime/blob/1316ae6695eec09cf47abdfd0bc6273aeaab1895/apis/common/v1/resource.go#L79

Could we use the existing `Policy` type from crossplane-runtime? Or are you trying to avoid introducing a `ResolutionPolicy` as well as a `ResolvePolicy`? (I regret those confusing names.)

Reply:
refactored to `Policy` type from crossplane runtime. Initially I avoided it due to ResolutionPolicy - considered that any provided reference should either be resolved or error emitted. With this change I also added check to not return error when fetching of env config fails AND resolution policy is `Optional`